src/modules/fetchMarkdown.ts

```diff
@@ -0,0 +1,49 @@
+import React from "react";
+
+const fetchMarkdown = (source: string, __previousSource: string, component: React.Component) => {
+	if (source === __previousSource) {
+		// don't fetch again if the source hasn't changed
+		return;
+	}
+
+	let url = require(`../articles/${source}`);
+
+	component.setState({
+		markdown: null,
+	});
+
+	fetch(url)
+		.then(resp => resp.text())
+		.then(text => {
+			component.setState({
+				markdown: text,
+				__previousSource: source,
+			});
+		});
+};
+
+// For use inside functional components.
+const fnFetchMarkdown = (
```

Comment:
We're duplicating functionality here. Can we rework this so we can just have one function, and make it a little more simple to use?